### DISCUSSION TITLE
Is there an alternative to dhcpcd-run-hooks to receive/query e.g. "domain_name_servers"

### DISCUSSION THREAD

**Discussion by mpflanzer** [rsmarples/dhcpcd#71] [Q&A] [answered]
Hi,

Is there a way to "connect" to a running `dhcpcd` process to query for instance the `domain_name_servers` it received from the DHCP server? I only found that I can get the property from the scripts that `dhcpcd` is executing via the run hooks. I was hoping that I could maybe also get notified or query it via the socket interfaces. But I couldn't find anything in the code. Am I just out of luck?

Cheers,
Moritz

**Comment by rsmarples** [comment, marked as answer]
`dhcpcd -U eth0 | grep domain_name_servers`

But see issue #69 as it might be broken in the latest release.

**Reply by mpflanzer** [reply]
Thanks for the fast answer and the heads up regarding the last release. :+1:
So to get notified it's only via the scripts but I can manually query the information via `dhcpcd -U`, correct?

**Reply by rsmarples** [reply]
Correct.

You have an both event and poll type notifications - what more do you want :)

**Reply by mpflanzer** [reply]
For the project I'm currently working on it would be a bit more comfortable to get an event within an already running program (which also launched dhcpcd). So before making use of the script events I just wanted to be sure I didn't overlook something that already exists. Except for that nothing more that I want. :wink:
